fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/ClientTransaction.java

```diff
@@ -98,24 +99,23 @@ public class ClientTransaction extends AbstractAuditableWithUTCDateTimeCustom {
     protected ClientTransaction() {}
 
     public static ClientTransaction payCharge(final Client client, final Office office, PaymentDetail paymentDetail,
-            final LocalDate transactionDate, final Money amount, final String currencyCode) {
+            final LocalDate transactionDate, final Money amount, final String currencyCode, final ExternalId externalId) {
         final boolean isReversed = false;
-        final String externalId = null;
         return new ClientTransaction(client, office, paymentDetail, ClientTransactionType.PAY_CHARGE.getValue(), transactionDate, amount,
                 isReversed, externalId, currencyCode);
     }
 
     public static ClientTransaction waiver(final Client client, final Office office, final LocalDate transactionDate, final Money amount,
             final String currencyCode) {
         final boolean isReversed = false;
-        final String externalId = null;
+        final ExternalId externalId = null;
```

Comment:
ExternalId by design should never be null. Please use ExternalId.empty() in this case.

Reply:
fixed